### PR TITLE
Remove starknet_in_rust

### PR DESCRIPTION
**Pull request by marioiordanov**
Closing #18 , #146 , #222 

## Development related changes

- Replaced imports of `starknet_in_rust` classes with `cairo_lang_starknet`. `starknet_in_rust` just reexported them.
- Remove usage of `starknet_in_rust` helper methods.
- Replaced with logic from `starknet_rs` for transactions hash computation
- `compute_casm_class_hash` logic copied from `starknet_in_rust`
- `compute_sierra_class_hash` - used `starknet`-rs implementation

## Note:
 tests using starknet_in_rust are temporary, only for checking if the copied logic have the same behavior as the original one. In one of the next PRs, when blockifier is updated, starknet_in_rust will be removed and the tests relying on it.

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [ ] Performed code self-review
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Documented the changes
- [ ] Linked the issues which this PR resolves
- [ ] Checked the TODO section in README.md if this PR resolves it
- [x] Updated the tests
- [x] All tests are passing - `cargo test`

### CODE REVIEW

File: crates/types/src/rpc/transactions/broadcasted_declare_transaction_v2.rs [hunk 1]
```diff
@@ -128,22 +129,23 @@ mod tests {
...
-        let sierra_class: SierraClass =
-            serde_json::from_value(serde_json::to_value(cairo_1_contract.clone()).unwrap())
-                .unwrap();
-        println!("{}", Felt::from(sierra_class.class_hash().unwrap()).to_prefixed_hex_str());
-
-        println!(
-            "{}",
-            Felt::from(compute_sierra_class_hash(&cairo_1_contract).unwrap()).to_prefixed_hex_str()
+
+        assert_eq!(
```
Comment (FabijanC) [inline crates/types/src/rpc/transactions/broadcasted_declare_transaction_v2.rs:141]: What happened in this test? Why are we only adding some assertions now? Also, are we keeping starknet_in_rust as a dependency for these couple of tests? Should the dependency then be labeled in Cargo.toml as used for testing only?

File: crates/types/src/rpc/contract_class.rs [hunk 2]
```diff
@@ -198,25 +193,114 @@ fn convert_sierra_to_codegen(
...
+        .map_err(|_| Error::ConversionError(ConversionError::InvalidFormat))?
+        .into())
+}
+
+/// Computes cairo_lang_starknet::CasmContractClass hash.
+/// Implementation copied from starknet_in_rust
+/// # Arguments
+///
+/// * `contract_class` - The contract class in casm format.
+pub fn compute_casm_class_hash(contract_class: &CasmContractClass) -> DevnetResult<Felt> {
```
Comment (FabijanC) [inline crates/types/src/rpc/contract_class.rs:228]: Why did we copy this function (it's only used in tests) if we are keeping starknet_in_rust as a dependency?

**Review by FabijanC — CHANGES_REQUESTED** [review]
Should #18, #146 and #222 be linked as closeable?

File: crates/types/src/rpc/transactions/broadcasted_declare_transaction_v2.rs (hunk 1 above)
Comment (marioiordanov) [reply crates/types/src/rpc/transactions/broadcasted_declare_transaction_v2.rs:141]: it is a dev-dependency in `types/Cargo.toml`

Comment (FabijanC) [reply crates/types/src/rpc/transactions/broadcasted_declare_transaction_v2.rs:141]: Ok, I was looking in `Cargo.toml` in project root. Resolving.

File: crates/starknet/src/starknet/mod.rs [hunk 3]
```diff
@@ -304,15 +304,29 @@ impl Starknet {
...
         block_context.fee_token_addresses.eth_fee_token_address =
             contract_address!(fee_token_address);
-        // inject cairo resource fee weights from starknet_in_rust
-        block_context.vm_resource_fee_cost =
-            std::sync::Arc::new(DEFAULT_CAIRO_RESOURCE_FEE_WEIGHTS.clone());
+        // copied cairo resource fee weights from starknet_in_rust
+        block_context.vm_resource_fee_cost = std::sync::Arc::new(HashMap::from([
+            ("n_steps".to_string(), N_STEPS_FEE_WEIGHT),
+            ("output_builtin".to_string(), 0.0),
+            ("pedersen_builtin".to_string(), N_STEPS_FEE_WEIGHT * 32.0),
```
[the remark this answers is not in the document]
Comment (marioiordanov) [reply crates/starknet/src/starknet/mod.rs:321]: blockifier crate has BlockContext::create_for_account_testing(), but its builtins are less compared to starknet_in_rust

**Comment by marioiordanov** [comment]
> Should #18, #146 and #222 be linked as closeable?

#222 was added in the development pane. The others will be added

File: crates/starknet/src/starknet/mod.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (FabijanC) [reply crates/starknet/src/starknet/mod.rs:321]: I asked on Telegram if there's anything we can import: https://t.me/sncorestars/34020

[the remark this answers is not in the document]
Comment (FabijanC) [reply crates/starknet/src/starknet/mod.rs:321]: Nobody is replying on Telegram, if you want to wait a bit more, you can. Feel free to resolve the issue then. I will now approve the PR.

**Review by FabijanC — APPROVED** [review]

**Pull request merged**